netcreateorg/netcreate-itest
# Comment edit lock

While a comment is being edited, we need to catch events to prevent
  the comment from inadvertently being closed:

  * prevent NodeTable or EdgeTable view/edit actions from triggering
    (handled by selection-mgr)
  * prevent NCNode from being able to click "Edit"
  * prevent NCCommentBtn close toggles (handled by NCCommentBtn)
  * prevent NCGraphRenderer events from selecting another node
    (handled by selection-mgr)

#### To Test

1. Open a node that already has a comment
2. Click the comment button to open the window (but don't edit or hit reply)
3. Try:
   - open NodeTable or EdgeTable and view or edit another node/edge -- you should be able to do it
   - in the selected node, click edit -- you should be able to edit the node
   - click the comment button -- you should be able to close the comment window
   - in the graph, click to select a node -- you should be able to select it (and close the prior comment)
4. Now add a new comment
   - open NodeTable or EdgeTable -- the view and edit buttons in the table should be hidden
   - in the selected node, the "Edit" button should be disabled so you can't click it
   - click the comment button -- nothing should happen -- you can't click the comment button to close the comment
   - in the graph, click to select a node -- nothing should happen -- you can't select another node
5. Save the comment -- everything should work again.
6. Add a reply -- everything should be locked again.

We probably want to add some messaging so that the caught clicks don't just quietly fail.  e.g. add an alert message informing the user to "Please finish editing the comment before selecting another comment" or something like that.
